### ISSUE TITLE
Routing allows for empty path value

### ISSUE THREAD

**Issue by aschempp** [contao/contao#264]
**Affected version(s)**

3.* (probably any), 4.*

**Description**  

While implementing #95 I stumbled upon a strange behavior of the routing implementation. If `useAutoItem` is disabled, a path of `foo/bar` will parse as alias `foo`, and an Input key `bar` without a value of *empty string* (due to `urldecode` converting `null` to string). I think thats not a correct behavior, the URL should not be valid.

**Comment by leofeyer** [comment]
As discussed in Mumble on January 17th, we want to support `foo/bar/.html` but return a 404 error for `foo/bar.html`.

**Comment by aschempp** [comment]
We could also solve it like this:

```
if (0 !== \count($fragments) % 2) {
    if (!$config->get('useAutoItem')) {
        throw new ResourceNotFoundException('Invalid number of arguments');

    }

    array_unshift($fragments, 'auto_item');
}
```

**Comment by leofeyer** [comment]
Fixed in Contao 4.4 in ecd00b8b3babc534bf7bc2521e1ee5a89eff0aa3.

**Comment by leofeyer** [comment]
Fixed in Contao 4.7 in 9f887cfdb2dcf83559ef95b183f6181097576b58.
